### ISSUE TITLE
two master hubs?

### ISSUE THREAD

**Issue by bingomanatee** [sirian/node-cluster-hub#5]
This may be a misuse on my part but can you have two master hubs? I'm trying something like that and getting cross signals. This may be a cognitive fail on my part but is there a legitimate use case for two master hubs with different listeners, owning different workers, or are you always supposed to have a single hub per process and use application logic to discriminate?

The use case is I have two sets of processes; one collects text for a record, the other splits that text into keywords and maintains a count of keywords.

**Comment by sirian** [comment]
Yes. you could provide `messageKey` argument when creating Hub. `let fooHub = new Hub('foo'); let barHub = new Hub('bar');`

**Comment by light24bulbs** [comment]
What about situations where you have a process manager and load balancer like pm2 which is handling your cluster, instead of node's internal clustering.  Would this project ever work or be able to be modified to work in that situation?

**Comment by sirian** [comment]
cluster hub is a wrapper for internal nodejs `cluster` module. So I don't understand what do you mean
